src/Browscap/Data/Validator/UseragentData.php

```diff
@@ -0,0 +1,219 @@
+<?php
+/**
+ * This file is part of the browscap package.
+ *
+ * Copyright (c) 1998-2017, Browser Capabilities Project
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types = 1);
+namespace Browscap\Data\Validator;
+
+use Browscap\Data\Helper\CheckDeviceData;
+use Browscap\Data\Helper\CheckEngineData;
+use Browscap\Data\Helper\CheckPlatformData;
+
+/**
+ * Class UseragentData
+ *
+ * @category   Browscap
+ *
+ * @author     Thomas Müller <mimmi20@live.de>
+ */
+class UseragentData
+{
+    /**
+     * @var \Browscap\Data\Helper\CheckDeviceData
+     */
+    private $checkDeviceData;
+
+    /**
+     * @var \Browscap\Data\Helper\CheckEngineData
+     */
+    private $checkEngineData;
+
+    /**
+     * @var \Browscap\Data\Helper\CheckPlatformData
+     */
+    private $checkPlatformData;
+
+    public function __construct()
+    {
+        $this->checkDeviceData   = new CheckDeviceData();
```

Comment:
These should be injected; also is this an entity? If so these shouldn't be injected at all; if you want to validate stuff, pass the entity into a validator, not the validator into the entity